rust/src/action/parquet_read/mod.rs

```diff
@@ -269,13 +272,27 @@ fn primitive_parquet_field_to_json_value(field: &Field) -> Result<serde_json::Va
 }
 
 fn convert_timestamp_millis_to_string(value: i64) -> Result<String, &'static str> {
+    let seconds = value / 1000;
+    let milliseconds = (value % 1000) as u32;
+
     let dt = Utc
-        .timestamp_opt(value / 1000, ((value % 1000) * 1000000) as u32)
+        .timestamp_opt(seconds, milliseconds * 1_000_000)
         .single()
         .ok_or("Value out of bounds")?;
     Ok(dt.to_rfc3339_opts(SecondsFormat::Millis, true))
 }
 
+fn convert_timestamp_micros_to_string(value: i64) -> Result<String, &'static str> {
+    let seconds = value / 1_000_000;
+    let microseconds = (value % 1_000_000) as u32;
+
+    let dt: chrono::DateTime<Utc> = Utc
+        .timestamp_opt(seconds, microseconds * 1_000)
+        .single()
+        .ok_or("Value out of bounds")?;
+    Ok(dt.to_rfc3339_opts(SecondsFormat::Micros, true))
```

Comment:
Do we want `SecondsFormat::Micros` here or match with Millisecond format as above?

Reply:
Seems like micros is fine. Spec specifically allows that one. https://github.com/delta-io/delta/blob/master/PROTOCOL.md#partition-value-serialization